TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -1537,8 +1537,54 @@ private Token ScanNumericLiteral(int startIndex)
 
                     if (BeSmartWithLevelNumber) { 
                         // Distinguish the special case of a LevelNumber
-                        if (tokensLine.ScanState.InsideDataDivision && tokensLine.ScanState.AtBeginningOfSentence) {
-                            token.CorrectType(TokenType.LevelNumber);
+                        if (tokensLine.ScanState.InsideDataDivision)
+                        {
+                            if (tokensLine.ScanState.AtBeginningOfSentence || GuessIfCurrentTokenIsLevelNumber())
+                            {
+                                token.CorrectType(TokenType.LevelNumber);
+                            }
+
+                            //This method is here to help recognize LevelNumbers when PeriodSeparator has been forgotten at the end of previous data definition.
+                            bool GuessIfCurrentTokenIsLevelNumber()
+                            {
+                                var lastSignificantToken = tokensLine.ScanState.LastSignificantToken;
+                                var beforeLastSignificantToken = tokensLine.ScanState.BeforeLastSignificantToken;
+
+                                bool currentTokenIsAtBeginningOfNewLine = token.Line == lastSignificantToken.Line + 1;
```

Comment:
What if there is an empty line between   tokenand lastSignificantToken?

Reply:
Good catch ! Fixed in latest commit.